ha/core/cluster/cluster_manager.py

```diff
@@ -296,3 +304,152 @@ def controller_list(self) -> list:
+                is to be returned.
+            **kwargs([dict]): Variable number of arguments that are used as filters,
+                e.g. "id" of the input "element".
+
+        Returns:
+            ([dict]): Returns dictionary. {"status": "Succeeded"/"Failed"/"Partial", "output": "", "error": ""}
+                status: Succeeded, Failed, Partial
+                output: Dictionary with element health status
+                error: Error information if the request "Failed"
+        """
+
+        try:
+            # Currently, only "id" supported in the variable arguments.
+            unsupported_element = True
+            for supported_element in CLUSTER_ELEMENTS:
+                if element == supported_element.value:
+                    unsupported_element = False
+                    break
+
+            # TODO: Fix below temporary code
+            if element != CLUSTER_ELEMENTS.NODE.value:
+                unsupported_element = True
+
+            if unsupported_element:
+                return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Invalid element"}
+            if kwargs and GET_SYS_HEALTH_ARGS.ID.value not in kwargs:
+                return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Invalid filter argument(s)"}
+
+            element_id = HEALTH_STATUSES.UNKNOWN.value
+            if kwargs:
+                element_id = kwargs["id"]
+
+            # Fetch the health status
+            system_health_controller = SystemHealthController(self._confstore)
+            self._status_dict = system_health_controller.get_status(CLUSTER_ELEMENTS.CLUSTER.value)
+
+            # Remove any keys which are not for the health status.
+            ignore_keys = []
+            for key in self._status_dict:
+                if "health" not in key:
+                    ignore_keys.append(key)
+            for key in ignore_keys:
+                del self._status_dict[key]
+
+            # Get the requested element level in the health hierarchy
+            self._health_hierarchy = HealthHierarchy()
+            element_level = self._health_hierarchy.get_element_level(element)
+
+            # TODO: Fix below
+            depth = self._health_hierarchy.get_total_depth()
+            # Prepare and return the output
+            output = StatusOutput(SYSTEM_HEALTH_OUTPUT_V1)
+            self.get_status(element, element_id = element_id, start_level = element_level, level = element_level, depth = depth, parent = output)
+            return output.to_json()
+        except Exception as e:
+            Log.error(f"Failed returning system health . Error: {e}")
+            return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Internal error"}
+
+    def get_status(self, element, element_id: str = HEALTH_STATUSES.UNKNOWN.value, start_level: int = 1, level: int = 1, depth: int = 1, parent: object = None):
```

Comment:
it should be private function. Is it possible to move this to system health module instead?
Cluster manager should have only exposed interfaces as much as possible.

Reply:
Please look at other such functions. In fact, all of the logic should go to system health and cluster_manager should be making a simple call to system health to get his information.

Reply:
Agree, will update in the refactor PR.